### PR TITLE
WT-13680 Improve error handling when Ruff is not found with the expected version

### PR DESCRIPTION
**Pull request by etienneptl**
The changes improve the error handling when a Ruff binary is found with a version that is not matching the expected one.

### CODE REVIEW

**Comment by etienneptl** [comment]
@lukech, I suggest your review the new code on its own instead of trying to compare before/after as I created new functions and shuffled things around.

File: dist/ruff_check.py [hunk 1]
```diff
@@ -1,48 +1,83 @@
...
+    show_ruff_installation_steps(ruff_expected_version)
     sys.exit(1)
 
-cmd = ["ruff", "check", "--fix", "../.", ".", "--config", ruff_config]
 
-if not run(cmd):
+cmd = ["ruff", "check", "--fix", "../.", ".", "--config", RUFF_TOML_FILE]
+
+if not run_ruff_checks(cmd):
     sys.exit(1)
```
Comment (lukech) [inline dist/ruff_check.py:83]: How about enclosing these lines of code in a `if __name__ == '__main__':` block to make it clear they will be executed when the script is called without argument (by s_all)?

**Review by lukech — APPROVED** [review]
LGTM % a minor suggestion.

File: dist/ruff_check.py (hunk 1 above)
Comment (etienneptl) [reply dist/ruff_check.py]: Sounds good - see a9ed3400c1696558c6be7821ed8ee690dca23d3f

Comment (lukech) [reply dist/ruff_check.py]: LGTM!

**Pull request merged**